[AV-68571] Vector Search - The Capella Files (#130)
couchbaselabs/docs-devex: modules/vector-search/pages/create-vector-search-index-ui.adoc

= Create a Vector Search Index in Quick Mode
:page-topic-type: guide
:description: Use Quick Mode to create a Vector Search index in Couchbase Capella.

[abstract]
{description}

> Use xref:create-search-index-ui.adoc[Advanced Mode] to add additional settings and more advanced options for your Vector Search index and xref:customize-index.adoc#type-mappings[type mappings].

Reviewer: Broken links (I think it's supposed to be in the `:search:` directory?)

Author: Thought I caught all these from my copy/paste - thanks!

> Use xref:create-search-index-ui.adoc[Advanced Mode] to add additional settings and more advanced options for your Vector Search index and xref:customize-index.adoc#type-mappings[type mappings].

Reviewer 2: incorrect links? both fail in the preview

Reviewer 2: which makes me wonder, does this quick mode topic also cover these advanced procedures? if not, maybe
After you create a VSI using quick made, you can configure more settings and add more advanced options for your Vector Search index and type mappings

You must create a Vector Search index before you can xref:run-vector-search-ui.adoc[run a search] that supports vector comparisons.

== Prerequisites 

* You have the Search Service enabled on a node in your database.
For more information about how to change Services on your database, see xref:cloud:clusters:modify-database.adoc[].

* You have a bucket with scopes and collections in your database. 
For more information, see xref:cloud:clusters:data-service/manage-buckets.adoc[].

> * You have documents in a keyspace inside that bucket that contain vector embeddings.

Reviewer: Nit: I tripped over this line a little bit. I'm wondering if we change "that bucket" to "your bucket" would help.

```suggestion
* You have documents in a keyspace inside your bucket that contain vector embeddings.
```

* You have logged in to the Couchbase Capella UI. 

== Procedure 

To create a Vector Search index with Quick Mode in Capella:

. On the *Databases* page, select the database where you want to create a Search index.
. Go to menu:Data Tools[Search].
. Click btn:[Create Search Index].
+

> By default, *Quick Mode* should be on.

Reviewer 2: is this where you might switch to Advanced Mode?

Author: Yes

. In the *Index Name* field, enter a name for the Vector Search index. 
+
[NOTE]
====
Your index name must start with an alphabetic character (a-z or A-Z). It can only contain alphanumeric characters (a-z, A-Z, or 0-9), hyphens (-), or underscores (_).

For version 7.6 and later, your index name must be unique inside your selected bucket and scope. You cannot have 2 indexes with the same name inside the same bucket and scope.
====
. Under *Type Mappings*, in the *Bucket* list, select the bucket that contains the documents you want to include in your index. 
. In the *Scope* list, select the scope that contains these documents. 
. Expand the collection that contains these documents. 
. In your document schema, select a field that contains your vector embeddings. 
+

> Vectors should be an array of floating point numbers.

Reviewer 2: maybe mention this in the prerequisites section?

Reviewer 2: also, is should the right word? are there workarounds if they are not floating point numbers, or in an array?

Author: No workaround - what would you suggest as an alternative?

Reviewer 2: Vectors must be represented as floating point numbers.

> . Configure the options for your new child field as follows:

Reviewer 2: my what now? I missed where a new field got created. Maybe this is evident when actually using the UI, but perhaps state that "select a field that contains your vector embeddings" in the previous step has the result of adding a field to the schema ?

Author: I added a note to the previous step - let me know if that resolves your question?

Reviewer 2: I guess I still don't understand if there's a new field (child or otherwise) being added or if I'm just adding configuration to something I selected.

Author: So, you have the document schema displayed. You select the field from the JSON schema that contains the data you want to be able to search from your index. 

That's the child field, the field you selected. 

Then, you're adding configuration onto that. 

You select the field on one side, and configure on the other.

.. In the *Type* list, select *vector*. 
.. In the *Dimension* field, check that the value matches the total number of elements in your vector field's array. 
+
The Search Service supports arrays up to 2048 elements.

> Capella should automatically fill in the dimension value for your selected array field.

Reviewer: Nit: perhaps make this a little less passive in tone—makes it sounds like it could break (I mean sure it can 😆 , but we don't want to convey that).

```suggestion
Capella automatically fills in the dimension value for your selected array field.
```

.. In the *Similarity metric* list, select the method to use to calculate the similarity between search term and Search index vectors.
+ 
For more information, see xref:search:quick-index-field-options.adoc#quick-field[Quick Field Type Mapping Options].
.. In the *Optimized for* list, select whether the Search Service should optimize Search queries for accuracy (*recall*) or speed (*latency*).
+

> For more information, see xref:xref:search:quick-index-field-options.adoc#quick-field[Quick Field Type Mapping Options].

Reviewer: ```suggestion
For more information, see xref:search:quick-index-field-options.adoc#quick-field[Quick Field Type Mapping Options].
```

.. Select *Index*.
. Click btn:[Submit].
. (Optional) Add additional collections or field type mappings to your index. 
. Click btn:[Create Index].

== Next Steps 

> This basic Vector Search index includes the vector embeddings from the field you specified in your type mapping.

Reviewer 2: is the type mapping what I did in step 9? 
is this from the specified field or from the "child field" that got created?

Author: Yes, this is the child field you created a type mapping for.

> If you chose to add additional child fields, the Search Service can also return data from those fields when you run a Vector Search query. 

Reviewer 2: the references to child fields are quite confusing to me. Does optional step 11 add these child fields?

Author: Yes

You can customize your Vector Search index like any other Search index to add additional data and improve search results. 
For more information about how to customize an index, see xref:search:customize-index.adoc[].

> CAUTION: Some Search index features are only available in Advanced Mode. If you edit your Search index in Advanced Mode, you cannot return to Quick Mode and keep any Advanced Mode settings. 

Reviewer 2: instead of "return to Quick Mode" I'd rephrase more like, 
After you use Advanced Mode to configure your Search Index, you cannot use Quick Mode for any further edits. Using Quick Mode resets all Advanced Mode settings to their default state.
...If that is, in fact, what happens...

= Create a Vector Search Index in Quick Mode
:page-topic-type: guide
:description: Use Quick Mode to create a Vector Search index in Couchbase Capella.

> 

Reviewer 3: Similar comment in the general FTS section--would be helpful to call out what Quick Mode doesn't give me...why/when would I choose Quick Mode over Advanced and vice versa?

You must create a Vector Search index before you can xref:search:run-vector-search-ui.adoc[run a search] that supports vector comparisons.

== Prerequisites 

* You have the Search Service enabled on a node in your database.
For more information about how to change Services on your database, see xref:cloud:clusters:modify-database.adoc[].

* You have a bucket with scopes and collections in your database. 
For more information, see xref:cloud:clusters:data-service/manage-buckets.adoc[].

* You have documents in a keyspace inside your bucket that contain vector embeddings.

* You have logged in to the Couchbase Capella UI. 

== Procedure 

To create a Vector Search index with Quick Mode in Capella:

. On the *Databases* page, select the database where you want to create a Search index.
. Go to menu:Data Tools[Search].
. Click btn:[Create Search Index].
+
By default, *Quick Mode* should be on.
. In the *Index Name* field, enter a name for the Vector Search index. 
+
[NOTE]
====
Your index name must start with an alphabetic character (a-z or A-Z). It can only contain alphanumeric characters (a-z, A-Z, or 0-9), hyphens (-), or underscores (_).

> For version 7.6 and later, your index name must be unique inside your selected bucket and scope. You cannot have 2 indexes with the same name inside the same bucket and scope.

Reviewer 3: Couchbase Server version7.6 or later

====

> . Under *Type Mappings*, in the *Bucket* list, select the bucket that contains the documents you want to include in your index. 

Reviewer 3: I originally intended to recommend using a different structure here, removing some of the conceptual or reference information from the steps. In the spirit of NOT documenting how to use the UI.

However, having read and reread this several times, I don't think you can avoid it. The procedure in the UI is just too complex/tricky.
